### PR TITLE
Fix in the bigquery input_tables missing

### PR DESCRIPTION
**Pull request by nizardeen**
Handled the BigQueryExtractor without returning it from None

### CODE REVIEW

**Comment by julienledem** [comment]
Hi @nizardeen, can you share an example of the airflow task you are running to reproduce this?

**Comment by julienledem** [comment]
FYI, when BigQuery caches the result, the input tables are not available in the API call.
The query cache is used only if you don't save the output of your query. (which happens if you are just testing things out and not saving the result)
See: https://cloud.google.com/bigquery/docs/cached-results#cache-exceptions
This can happen when testing things out but will not happen in production.

File: integrations/airflow/marquez_airflow/extractors/bigquery_extractor.py [hunk 1]
```diff
@@ -115,16 +115,29 @@ def extract_on_complete(self, task_instance) -> [StepMetadata]:
...
+
     def _get_input_from_bq(self, job, context, source, client):
-        if not job._properties.get('statistics')\
+        
+        if job._properties.get('statistics').get('query').get('referencedTables') is not None:
+            bq_input_tables = job._properties.get('statistics')\
+            .get('query')\
+            .get('referencedTables')
+
+        elif context['bigquery.sql.parsed.inputs'] is not None:
```
Comment (wslulciuc) [inline integrations/airflow/marquez_airflow/extractors/bigquery_extractor.py:131]: The main usage of the context key `bigquery.sql.parsed.inputs` is to compare the input and output tables returned via the `BigQuery` client call. For `BigQueryOperator`, we use the job details object (see example [here](https://github.com/MarquezProject/marquez/blob/main/integrations/airflow/tests/extractors/job_details.json)) returned by the client to set the in / out tables, so we wouldn't want to default to using the context here (though that might be an option in the future).

Note that the context key also helps to improve the [SQL parser](https://github.com/MarquezProject/marquez/blob/main/integrations/airflow/marquez_airflow/extractors/sql/experimental/parser.py) shipped with the `marquez-airflow` lib if they differ from wants returned by `BigQuery`

**Comment by julienledem** [comment]
@nizardeen Is this still needed? The implementation would need to evolve.

**Comment by wslulciuc** [comment]
Thanks for opening this PR, @nizardeen! For now, I'm going to close this PR as it's a bit out of date and looks like some clarity on the issue is still needed before we can merge. Feel free to reopen and/our reach out to us so we can iterate on your implementation.

**Pull request closed without merging**